lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -250,7 +285,11 @@ sub _check_error {
     return (400, 'Subfolder has invalid characters') if $subfolder && $subfolder =~ m!/!;
     return (400, 'Filename has invalid characters')  if $filename  && $filename  =~ m!/!;
 
+    my $batch;
+    ($project, $batch) = split(/\|/, $project, 2);
     return (404, 'Invalid Project {' . $project . '}') if $project && !-d Mojo::File->new($home, $project);
+    return (404, 'Invalid Batch {' . $project . '/' . $batch . '}')
```

Comment:
I get that you're using `|` as separator because it must be a character which is allowed in directory names. But then I'd use `|` for consistency here, too.